src/kili/core/graphql/graphql_client.py

```diff
@@ -238,38 +242,43 @@ def _execute(
                     },
                     **kwargs,
                 )
-            except (exceptions.TransportQueryError, graphql.GraphQLError) as err:
+            except (
+                graphql.GraphQLError,
+                exceptions.TransportError,
+                exceptions.TransportQueryError,
+            ) as err:
                 if isinstance(err, exceptions.TransportQueryError):
                     raise GraphQLError(error=err.errors) from err
                 if isinstance(err, graphql.GraphQLError):
                     raise GraphQLError(error=err.message) from err
-            return result  # type: ignore
+                raise GraphQLError(error=err) from err
+            return result
 
         document = query if isinstance(query, DocumentNode) else gql(query)
 
-        ret = {}
         try:
-            ret = _execute(document, variables, **kwargs)
-        except (GraphQLError, TransportServerError) as err:
-            if isinstance(err, GraphQLError):
-                # if error is due do parsing or local validation of the query (graphql.GraphQLError)
-                # we refresh the schema and retry once
-                if isinstance(err.__cause__, graphql.GraphQLError):
-                    self._purge_graphql_schema_cache_dir()
-                    self._gql_client = self._initizalize_graphql_client()
-                    ret = _execute(
-                        document, variables, **kwargs
-                    )  # if it fails again, we crash here
-                else:
-                    raise err
-            if isinstance(err, TransportServerError):
-                if err.code == 401:
-                    # if error is due to authentication, we retry once
-                    time.sleep(1)
-                    ret = _execute(document, variables, **kwargs)
-                else:
-                    raise err
-        return ret
+            return _execute(document, variables, **kwargs)
+        except GraphQLError as err:
+            # if error is due do parsing or local validation of the query (graphql.GraphQLError)
+            # we refresh the schema and retry once
+            if isinstance(err.__cause__, graphql.GraphQLError):
+                self._purge_graphql_schema_cache_dir()
+                self._gql_client = self._initizalize_graphql_client()
+                return _execute(document, variables, **kwargs)  # if it fails again, we crash here
+
+            # if backend returned a 401 error, we retry a few times
```

Comment:
if the api key is wrong, this means that the user will have to wait 30 seconds before being notified?

Reply:
if the api key is wrong, it'll be caught during the Kili() init in Kiliauth